### ISSUE TITLE
Running agent on Mac OS X

### ISSUE THREAD

**Issue by prawana-perera** [awslabs/amazon-kinesis-agent#163]
We have a mix of servers/desktops across Windows, Linux and Mac OS X. I see there is Windows agent support (in addition to Linux) and was hoping to get agent running on Mac OS X as well. 

I am happy to try to get it working on Mac (i.e. customising the setup script to work on Mac) but wanted to check if this is:
- Feasible? 
- A good idea? 
- Any inputs from anyone who may have done this, have experience with this or are looking to do this?

Thanks.

**Comment by willUrgently** [comment]
Good idea. Platform security applies to all platforms. Even Macs or SGI boxes.

Feasible? My take is that the build is a server build. And as such should be BSD compatible, Not the framework that replaced Cocoa...or whatever the new scripting language is. (They've got it on iOS and OS X I think)

so its a darwin_x64 that we are talking about? 

I can help porting the socket performance code. Agent could use adaptive socket parametric groups or equations ... i don't know. Takes forever to load the first big payload, and forever to send the last little one.

**Comment by pkava** [comment]
I tried installing agent on mac(Mojave). It gives following error (it doesnt detect correct OS) :  

sudo ./setup --install
Uninstalling aws-kinesis-agent ...
Removing aws-kinesis-agent from system services...
./setup: line 163: userdel: command not found
./setup: line 164: groupdel: command not found
Installing Kinesis Agent ...
./setup: line 95: yum: command not found

**Comment by pospospos2007** [comment]
I got the same error on Mac 13.4.1

Detected OS Distro: RedHat
Uninstalling aws-kinesis-agent ...
Removing aws-kinesis-agent from system services...
./setup: line 166: userdel: command not found
./setup: line 167: groupdel: command not found
Installing Kinesis Agent ...
./setup: line 95: yum: command not found
